pennylane/pulse/rydberg_hamiltonian.py

```diff
@@ -218,29 +218,44 @@ def circuit_local(params):
             qml.evolve(H)(params, t=[0, 10])
             return qml.expval(qml.PauliZ(0))
 
-    >>> params = [2.4, [1.3, -2.0]]
+    >>> params = [2.4, [1.3, -2.0], -1.5]
     >>> circuit_local(params)
-    Array(0.45782223, dtype=float64)
+    Array(0.80238028, dtype=float64)
     >>> jax.grad(circuit_local)(params)
-    [Array(-0.33522988, dtype=float64),
-     [Array(0.40320718, dtype=float64, weak_type=True),
-      Array(-0.12003976, dtype=float64, weak_type=True)]]
+    [Array(-0.73273312, dtype=float64),
+     [Array(-0.45635171, dtype=float64, weak_type=True),
+      Array(0.76572817, dtype=float64, weak_type=True)],
+     Array(0.01027312, dtype=float64)]
     """
     if isinstance(wires, int):
         wires = [wires]
+    trivial_detuning = detuning is None or (
+        not callable(detuning) and qml.math.isclose(detuning, 0.0)
```

Comment:
Does this condition preserve JIT-compatibility when branching logically on it further below? :thinking: It seems that the tests cover this, so should be okay, but I'm actually positively surprised by that :D

Reply:
Hmmm. I don't think any of the integration tests verified any cases where either the amplitude or detuning are trivial. I will add a couple of tests just to be sure, but I'm quite glad that nothing fails so far 🤞 . JIT-ing completely slipped my mind. Thanks for pointing it out!

Reply:
I think this is okay because this constructor is usually called outside the jitting scope?